src/juliapkg/deps.py

```diff
@@ -30,6 +30,13 @@ def save_meta(meta):
     assert meta.get('meta_version') == META_VERSION
     fn = STATE['meta']
     os.makedirs(os.path.dirname(fn), exist_ok=True)
+    if os.path.exists(fn):
+        with open(fn) as fp:
+            old_meta_json = fp.read()
+        meta_json = json.dumps(meta)
+        if meta_json == old_meta_json:
+            # No need to write out if nothing changed
```

Comment:
This seems reasonable, but can you add [`Path(fn).touch()`](https://docs.python.org/3/library/pathlib.html#pathlib.Path.touch) just before `return`?

We use the mtime on the meta file as a quick indicator of whether we can skip resolving dependencies, so if we don't update the mtime when we resolve, we might mistakenly resolve again next time around.

Reply:
Can you point e to where the mtime of the meta file is used? I only found it being used for the deps files.

Reply:
No, because I was mistaken. It's a while since I looked at this.